be/src/agent/task_worker_pool.cpp

```diff
@@ -414,19 +415,26 @@ void* TaskWorkerPool::_drop_tablet_worker_thread_callback(void* arg_this) {
         std::vector<std::string> error_msgs;
         TStatus task_status;
 
-        auto dropped_tablet = StorageEngine::instance()->tablet_manager()->get_tablet(drop_tablet_req.tablet_id);
-        if (dropped_tablet != nullptr) {
-            TabletDropFlag flag = force_drop ? kDeleteFiles : kMoveFilesToTrash;
-            auto st = StorageEngine::instance()->tablet_manager()->drop_tablet(drop_tablet_req.tablet_id, flag);
-            if (!st.ok()) {
-                LOG(WARNING) << "drop table failed! signature: " << agent_task_req.signature;
-                error_msgs.emplace_back("drop table failed!");
-                status_code = TStatusCode::RUNTIME_ERROR;
+        auto st = StorageEngine::instance()->tablet_manager()->get_tablet(drop_tablet_req.tablet_id);
+        if (!st.ok()) {
+            LOG(WARNING) << "failed to get tablet!";
+            error_msgs.emplace_back("get table failed!");
+            status_code = TStatusCode::RUNTIME_ERROR;
+        } else {
+            auto dropped_tablet = st.value();
+            if (dropped_tablet != nullptr) {
```

Comment:
According your implementation of `TabletManager::get_tablet`, if `st.ok()` is true, `st.value()` must not be NULL, right?

Reply:
if st.ok()., it means  tablet meanager successfully get the tablet ,and the st.value() is not null.   
if to_status(st).is_not_found, it means the tablet is not found in rocksdb and tabletmetacache.
for the other err status , it means that tablet manager failed to search rocksdb or other questions.